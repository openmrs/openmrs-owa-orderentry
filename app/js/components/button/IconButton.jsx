import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class IconButton extends PureComponent {
  static propTypes = {
    iconClass: PropTypes.string,
    iconTitle: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    dataContext: PropTypes.object,
    icon: PropTypes.string,
    id: PropTypes.string,
  }

  static defaultProps = {
    iconClass: '',
    iconTitle: '',
    dataContext: {},
    icon: null,
    id: '',
  }

  handleClick = (event) => {
    event.preventDefault();
    event.stopPropagation();
    this.props.onClick(this.props.dataContext);
  }

  render() {
    const {
      id, iconClass, iconTitle, icon,
    } = this.props;
    return (
      <span>
        <a id={`${id}`} href="#" onClick={this.handleClick}>
          <i className={`${iconClass} scale`} title={iconTitle}>{icon && icon}</i>
        </a>
      </span>
    );
  }
}

export default IconButton;
