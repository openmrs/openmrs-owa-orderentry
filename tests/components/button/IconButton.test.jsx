import React from 'react';
import IconButton from '../../../app/js/components/button/IconButton';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<IconButton {...props} />);
  }
  return mountedComponent;
};

describe('Component: IconButton', () => {
  beforeEach(() => {
    props = {
      iconClass: 'icon-remove',
      iconTitle: 'Header',
      onClick: jest.fn(),
      id:'icon-btn-anchor',
      icon: '&#x25B2;'
    };
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('has a handler for click events', () => {
    const event = {
      preventDefault: () => {},
      stopPropagation: () => {},
    };
    jest.spyOn(event, 'preventDefault');
    jest.spyOn(event, 'stopPropagation');
    const renderedComponent = getComponent();
    const anchor = renderedComponent.find('#icon-btn-anchor');
    const handleClick = jest.spyOn(props, 'onClick');
    anchor.simulate('click', event);
    expect(handleClick).toBeCalled();
    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();
  });
});
