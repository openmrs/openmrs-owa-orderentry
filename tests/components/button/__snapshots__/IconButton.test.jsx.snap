// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IconButton renders properly 1`] = `
<span>
  <a
    href="#"
    id="icon-btn-anchor"
    onClick={[Function]}
  >
    <i
      className="icon-remove scale"
      title="Header"
    />
  </a>
</span>
`;
